akka-docs-dev/rst/stages-overview.rst

```diff
@@ -0,0 +1,134 @@
+
+All stages by default backpressure if the computation they encapsulate is not fast enough to keep up with the rate of
+incoming elements from the preceding stage. There are differences though how the different stages handle when some of
+their downstream stages backpressure them. This table provides a summary of all built-in stages and their semantics.
+
+All stages stop and propagate the failure downstream as soon as any of their upstreams emit a failure unless supervision
+is used. This happens to ensure reliable teardown of streams and cleanup when failures happen. Failures are meant to
+be to model unrecoverable conditions, therefore they are always eagerly propagated.
+For in-band error handling of normal errors (dropping elements ifa map fails for example) you should use the
+upervision support, or explicitly wrap your element types in a proper container that can express error or success
+states (for example `Try` in Scala).
+
+Custom components are not covered by this table since their semantics are defined by the user.
+
+Simple processing stages
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+These stages are all expressible as a `PushPullStage`. These stages can transform the rate of incoming elements
+since there are stages that emit multiple elements for a single input (e.g. `mapConcat') or consume
+multiple elements before emitting one output (e.g. `filter`). However, these rate transformations are data-driven, i.e. it is
+the incoming elements that define how the rate is affected. This is in contrast with :ref:`detached-stages-overview`
+which can change their processing behavior depending on being backpressured by downstream or not.
+
+=====================  =========================================================================================================================   ==============================================================================================================================  =====================================================================================
+Stage                  Emits when                                                                                                                  Backpressures when                                                                                                              Completes when
+=====================  =========================================================================================================================   ==============================================================================================================================  =====================================================================================
+map                    the function mapping the element finishes                                                                                   downstream backpressures                                                                                                        Upstream completes
```

Comment:
“when the function finishes” might be misunderstood or misleading; perhaps “when the mapping function returns an element”?